### PR TITLE
Design tokens for email

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This PR adds a new "target" for the design tokens pipeline, specifically to support the work that @venusang is doing in the context of the "adoption" of the HDS visual language in the email templating system.

Since the email templates don't support CSS variables, it was necessary to export the design tokens as Sass variables, so they can then be consumed by the Foundation Email framework as "overrides" of their defaults.

We also export CSS helper classes, in case they have to be used in the email templating system, or by the consumers of the templating system itself.

_Notice: for this initial release, we will focus only on foundations; later we may want to include design tokens for the `Button` component too, but will be discussed and decided in a second phase (could also be a fast-follow, it depends on the explorations @venusang is doing on her side of the code)._

### :hammer_and_wrench: Detailed description

In this PR I have:
- added new pipeline to style dictionary for `cloud-email` - `email/sass-variables` to be used in email templating
- generated new design tokens (as Sass variables) for email templating

### :link: External links

Main HDS epic: https://hashicorp.atlassian.net/browse/HDS-2158
Jira ticket: https://hashicorp.atlassian.net/browse/HDS-3158

Other relevant links:
- [PLAT-713: HDS Adoption/Migration for Identity Flows](https://hashicorp.atlassian.net/browse/PLAT-713)
- [HCPIE-742: Q1 HDS Adoption (Identity Emails)](https://hashicorp.atlassian.net/browse/HCPIE-742)
- [[HCP-010] HDS Adoption Strategy for cloud-email](https://docs.google.com/document/d/1pkIzEW9VFWsmfz5VPP4BWSvSQo7cTwIGvd2-LZuCG0o/edit)

***

### 👀 Component checklist

- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by venusang — APPROVED** [review]
✨ so far working like a charm!  Thanks so much for this.

File: packages/tokens/scripts/build.ts [hunk 1]
```diff
@@ -85,6 +85,12 @@ StyleDictionaryPackage.registerTransformGroup({
     transforms: ['attribute/cti', 'name/cti/kebab', 'font-size/rem', 'size/px', 'color/css', 'color/with-alpha', 'time/seconds']
 });
 
+StyleDictionaryPackage.registerTransformGroup({
+    name: 'products/email',
+    // TODO! update this according to the needs we have for the email templates
+    transforms: ['attribute/cti', 'name/cti/kebab', 'font-size/rem', 'size/px', 'color/css', 'color/with-alpha', 'time/seconds']
```
Comment (didoo) [inline packages/tokens/scripts/build.ts:91]: @venusang do you need the `font-size` properties expressed in `rem` or in `px`?

**Comment by venusang** [comment]
👋 @didoo - it looks like we need `token-elevation-low-box-shadow` for the buttons - does that sound right to you?

File: packages/tokens/dist/cloud-email/helpers/color.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Do not edit directly
+ * Generated on Thu, 29 Feb 2024 14:07:07 GMT
+ */
+
+.hds-border-primary { border: 1px solid var(--token-color-border-primary); }
```
Comment (venusang) [inline packages/tokens/dist/cloud-email/helpers/color.css:6]: I think I spoke too soon.... it doesn't look like Foundation for Emails is honoring the `var()` functions that are used by the hds classes.

Uploading CleanShot 2024-03-04 at 17.33.28.mp4…

File: packages/tokens/scripts/build.ts (hunk 1 above)
Comment (didoo) [reply packages/tokens/scripts/build.ts]: never mind, went ahead and changed to `px` (by the look of it seems [Foundation](https://get.foundation/emails/docs/typography.html#sass-reference) uses `px`)

**Review by KristinLBradley — APPROVED** [review]

**Comment by didoo** [comment]
> Do we want to document these sets of APIs we're exposing? I know we're touching a bit on the developer platform in [our getting started guide](https://helios.hashicorp.design/getting-started/for-engineers#install-tokens-package), but we also have marketing and now `cloud-ui` email.

@alex-ju good point, tracked in https://hashicorp.atlassian.net/browse/HDS-3201

**Pull request merged**